Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -831,12 +842,13 @@ public struct BSONObjectID: BSONValue, Equatable, CustomStringConvertible, Codab
         self.oid = oid
     }
 
-    /// Initializes an `BSONObjectID` from the provided hex `String`. Returns `nil` if the string is not a valid
-    /// ObjectID.
+    /// Initializes an `BSONObjectID` from the provided hex `String`.
+    /// - Throws:
+    ///   - `InvalidArgumentError` if string passed is not a valid BSONObjectID
     /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
-    public init?(_ hex: String) {
+    public init(_ hex: String) throws {
         guard bson_oid_is_valid(hex, hex.utf8.count) else {
-            return nil
+            throw InvalidArgumentError(message: "Cannot create ObjectId from \(hex)")
```

Comment:
did we ever decide what we were going to do with BSON errors? As written here, we'd need to catch and rethrow any BSON errors as driver errors.

Reply:
I thought we decided on catch and rethrow but I could be remembering wrong.

Reply:
I think catch and rethrow is what we gotta do to make swift-bson also be its own standalone library if that use case arises